### ISSUE TITLE
Ghost in the shell segfault

### ISSUE THREAD

**Issue by iota97** [hrydgard/ppsspp#12752]
After #12741 Ghost in the shell segfault at start up.

Placing a return before this code/removing it make it goes is game (or at least enought to show the intro logo).
```
// Alright, we have another to call.
DEBUG_LOG(HLE, "Executing HLE mips call at %08x, sp=%08x", currentMIPS->pc, sp);
currentMIPS->pc = stackData->func;
currentMIPS->r[MIPS_REG_RA] = HLEMipsCallReturnAddress();
for (int i = 0; i < (int)stackData->argc; i++) {
	currentMIPS->r[MIPS_REG_A0 + i] = Memory::Read_U32(sp + sizeof(HLEMipsCallStack) + i * sizeof(u32));
}
```
Debug log: Executing HLE mips call at 08000038, sp=08b94220

**Comment by unknownbrackets** [comment]
The logging was wrong - fixed now.

It should now log the correct address now.  Before that change, was it calling all the same addresses with all the same args?  Somewhere must be different, I guess...

-[Unknown]

**Comment by unknownbrackets** [comment]
Another idea - I didn't necessarily pay attention to the order the funcs were called from sceFont before in the old way.  Now they are called in the same order they're enqueued.

Maybe the order changed and that's the problem?

-[Unknown]

**Comment by iota97** [comment]
The last 2 call have the same PC, is this supposed to happen?

```
Flushing 2 HLE mips calls from sceFontClose, sp=08b94250
Executing HLE mips call at 0897b674, sp=08b94200
Executing next HLE mips call at 0897b674, sp=08b94220
Segmentation fault (core dumped)
```
PS: There is a way to get the log from hle.cpp in the console?

**Comment by unknownbrackets** [comment]
You can change `#if PPSSPP_PLATFORM(UWP) && defined(_DEBUG)` in Common/LogManager.cpp to `||`.

I think that's correct, no?  Both `CloseFont()` and `flushFont()` call the same `freeFuncAddr()`, so both those values are the free func.  `sp` is different though, which represents the args.

If you comment out the enqueue in `flushFont()`, or the other one in `CloseFont()`, does it prevent the crash?

Maybe also try changing the logging from:

```c++
DEBUG_LOG(HLE, "Executing HLE mips call at %08x, sp=%08x", currentMIPS->pc, sp);
```

To:

```c++
DEBUG_LOG(HLE, "Executing HLE mips call at %08x, sp=%08x, a0=%08x, a1=%08x", currentMIPS->pc, sp, currentMIPS->r[MIPS_REG_A0], currentMIPS->r[MIPS_REG_A1]);
```

And same for "next HLE mips call" (but have to move it to the end of the func so those values are populated.)  That would confirm that a1 is correctly different between the two.

-[Unknown]

**Comment by iota97** [comment]
Commenting either one (or both) prevent the crash (it never log to a "next HLE misp call", so I guess is just never get to that code).

Without commenting anything that's the log:

```
Executing HLE mips call at 0897b674, sp=08b94200, a0=0903f700, a1=08b4a2f0
Executing next HLE mips call at 0897b674, sp=08b94220, a0=0903f700, a1=08b6dcb0
Segmentation fault (core dumped)
```

**Comment by unknownbrackets** [comment]
Hm, is that the same order it had before the HLE change that caused the crash?  Maybe we try putting flushFont() higher?  Maybe I could create a test to see what order is correct on real firmware...

Sorry, I don't have this game and haven't been able to find another font/mpeg/etc. game crashing...

-[Unknown]

**Comment by iota97** [comment]
I'm not sure what you mean with higher, I tried moving it at start of CloseFont(), still crash the same.
About the previous order I'm not sure how to check that as well, I can see the last call before it crash becouse they are the last, but I'm having trouble seeing them when all goes well.

PS (may be totally irrelevant) also commenting action->run(mc) like this make it go in game again.
```
MipsCall mc;
mc.savedV0 = finalMarker->v0;
mc.savedV1 = finalMarker->v1;
//action->run(mc);
finalMarker->v0 = mc.savedV0;
finalMarker->v1 = mc.savedV1;
```

**Comment by unknownbrackets** [comment]
If you turn off fast memory in settings, is it jit/game code crashing, or is the crash in C++?  What's the stack trace if you run it through the debugger?

What happens if you comment out `hleEnqueueCall(fontLib->allocFuncAddr(), 2, args, action);`?  That's the only action enqueuing, just wondering if that's the `action->run(mc);` that causes the crash.  Though I have games successfully triggering that...

-[Unknown]

**Comment by iota97** [comment]
Well, fast memory off make it actually go in game so I guess C++ is fine:
```
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:92 ReadFromHardware: Invalid address 1eadbeeb
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:92 ReadFromHardware: Invalid address 1eadbeeb
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:128 WriteToHardware: Invalid address 1eadbeeb
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:92 ReadFromHardware: Invalid address 1eadbee7
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:92 ReadFromHardware: Invalid address 1eadbeef
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:92 ReadFromHardware: Invalid address 1eadbef3
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:128 WriteToHardware: Invalid address 00000008
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:128 WriteToHardware: Invalid address 0000000c
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:92 ReadFromHardware: Invalid address 1eadbeeb
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:92 ReadFromHardware: Invalid address 1eadbeef
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:92 ReadFromHardware: Invalid address 1eadbef3
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:128 WriteToHardware: Invalid address 00000008
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:128 WriteToHardware: Invalid address 0000000c
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:128 WriteToHardware: Invalid address 1eadbeeb
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:128 WriteToHardware: Invalid address 1eadbee7
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:128 WriteToHardware: Invalid address 1eadbef3
59:47:181 KokakuGame   W[MEMMAP]: Core/MemMapFunctions.cpp:128 WriteToHardware: Invalid address 1eadbeef
59:47:181 KokakuGame   I[SCEFONT]: HLE/sceFont.cpp:962 sceFontDoneLib(08b491d0)
```
About the stack trace I'll try to get one (never done before TBH).

Commenting `hleEnqueueCall(fontLib->allocFuncAddr(), 2, args, action);` also fix the crash.

**Comment by unknownbrackets** [comment]
I forgot completely to test without skip deadbeef filling.  Probably fixed by 8b6d2f9.

-[Unknown]

**Comment by iota97** [comment]
Fixed indeed.
